File: car-showroom-python/car.py
class Car:
    def __init__(self, car_id: int, make: str, model: str, year: int, price: float):
        if not make or not make.strip():
            raise ValueError("Make cannot be blank")
        if not model or not model.strip():
            raise ValueError("Model cannot be blank")
        if year < 1900 or year > 2100:
            raise ValueError("Year must be between 1886 and 2100")
        if price < 0:
            raise ValueError("Price cannot be negative")

        self._id = car_id
        self._make = make
        self._model = model
        self._year = year
        self._price = price
        self._available = True

    @property
    def id(self):
        return self._id

    @property
    def make(self):
        return self._make

    @make.setter
    def make(self, value: str):
        if not value or not value.strip():
            raise ValueError("Make cannot be blank")
        self._make = value

    @property
    def model(self):
        return self._model

    @model.setter
    def model(self, value: str):
        if not value or not value.strip():
            raise ValueError("Model cannot be blank")
        self._model = value

    @property
    def year(self):
        return self._year

    @year.setter
    def year(self, value: int):
        if value < 1900 or value > 2100:
            raise ValueError("Year must be between 1886 and 2100")
        self._year = value

    @property
    def price(self):
        return self._price

    @price.setter
    def price(self, value: float):
        if value < 0:
            raise ValueError("Price cannot be negative")
        self._price = value

    @property
    def available(self):
        return self._available

    @available.setter
    def available(self, value: bool):
        self._available = value

    def __str__(self):
        return (
            f"Car[id={self._id}, make={self._make}, model={self._model}, "
            f"year={self._year}, price={self._price}, available={self._available}]"
        )

File: car-showroom-java/src/main/java/com/showroom/Car.java
package com.showroom;

public class Car {

    private final int id;
    private String make;
    private String model;
    private int year;
    private double price;
    private boolean available;

    public Car(int id, String make, String model, int year, double price) {
        if (make == null || make.isBlank()) throw new IllegalArgumentException("Make cannot be blank");
        if (model == null || model.isBlank()) throw new IllegalArgumentException("Model cannot be blank");
        if (year < 1886 || year > 2100) throw new IllegalArgumentException("Invalid year: " + year);
        if (price < 0) throw new IllegalArgumentException("Price cannot be negative");

        this.id = id;
        this.make = make;
        this.model = model;
        this.year = year;
        this.price = price;
        this.available = true;
    }

    public int getId() { return id; }
    public String getMake() { return make; }
    public String getModel() { return model; }
    public int getYear() { return year; }
    public double getPrice() { return price; }
    public boolean isAvailable() { return available; }

    public void setMake(String make) {
        if (make == null || make.isBlank()) throw new IllegalArgumentException("Make cannot be blank");
        this.make = make;
    }

    public void setModel(String model) {
        if (model == null || model.isBlank()) throw new IllegalArgumentException("Model cannot be blank");
        this.model = model;
    }

    public void setYear(int year) {
        if (year < 1886 || year > 2100) throw new IllegalArgumentException("Invalid year: " + year);
        this.year = year;
    }

    public void setPrice(double price) {
        if (price < 0) throw new IllegalArgumentException("Price cannot be negative");
        this.price = price;
    }

    public void setAvailable(boolean available) { this.available = available; }

    @Override
    public String toString() {
        return String.format("Car{id=%d, make='%s', model='%s', year=%d, price=%.2f, available=%s}",
                id, make, model, year, price, available);
    }
}
